File: tapish/leetcode/RemoveElement.cpp
//https://leetcode.com/problems/remove-element/
//O(n) time, O(1) space
// For each array element, determine its position in the final array, where all "val" elements are deleted
int removeElement(vector<int>& nums, int val) {
    int size = nums.size();
    int count = 0;

    for (int i = 0; i < size; i++) {
        if (nums[i] == val) {
            count++;
        }
        else {
            nums[i - count] = nums[i];
        }
    }
    return size - count;
}

File: priya/leetcode/Remove Element.py
def removeElement(self, nums, val):
        n = len(nums)

        for i in range(n):
            while(i<n and nums[i]==val):
                nums[i] = nums[n-1]
                n-=1
        return n

# Question at : https://leetcode.com/problems/remove-element/#/description
